src/frontend/javascript/event-tracking.js

```diff
@@ -0,0 +1,96 @@
+function EventTracking () {
+  this.$bodyElement = document.querySelector('body')
+  this.$mainHeading = this.$bodyElement.querySelector('h1')
+  this.errorStatusCode,
+  this.isErrorPage = this.$mainHeading ? 
+    this.$mainHeading.textContent === 'Page not found' ? 
+    this.errorStatusCode = '404' : this.$mainHeading.textContent === 'Sorry an error occurred' ? 
+    this.errorStatusCode = '500' : this.$mainHeading.textContent === 'Page not authorised' ? 
+    this.errorStatusCode = '403': undefined : false
+ }
+
+EventTracking.prototype.init = function() {
+
+  this.$bodyElement.addEventListener('click', this.handleClickEvent.bind(this))
+
+  if (this.isErrorPage) {
+    this.errorPageEvent(this.errorStatusCode)
+  }
+}
+
+EventTracking.prototype.handleClickEvent = function (e) {
```

Comment:
Since we're in a Typescript project, we should make use of the type annotations

Reply:
this is client-side JS so we're not shipping TS or compiling from es6 to es5

Reply:
Aaah! Ok, gotcha.